### PR TITLE
use custom file format for polygon tests

### LINKED ISSUE #910: Fix Windows CI
For some reason #895 caused our Windows CI to start inexplicably failing on two tests: `Polygon.Zebra` and `Polygon.Zebra3` with `Process exited with code 127`. It's a pretty opaque error, but 100% repro on Windows, 0% on all other platforms. For now we have a workaround - details in this [thread](https://github.com/elalish/manifold/pull/895#issuecomment-2323857439), but we should fix the root cause before release.

### PR DESCRIPTION
**Pull request by pca006132**
Fixes #910

see https://github.com/elalish/manifold/pull/911#issuecomment-2327446700

### CODE REVIEW

**Comment by pca006132** [comment]
Probably need to use https://emscripten.org/docs/porting/files/packaging_files.html#packaging-files for wasm. Will try it later tonight.

File: test/polygons/conversion.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Usage:
+# python conversion.py file.cpp output.txt
```
Comment (elalish) [inline test/polygons/conversion.py:2]: Do we need this script anymore if all the old test files have already been converted?

File: src/polygon/src/polygon.cpp [hunk 2]
```diff
@@ -123,7 +123,7 @@ void CheckGeometry(const std::vector<ivec3> &triangles,
 
 void Dump(const PolygonsIdx &polys) {
   for (auto poly : polys) {
-    std::cout << "polys.push_back({" << std::setprecision(9) << std::endl;
+    std::cout << "polys.push_back({" << std::setprecision(15) << std::endl;
```
Comment (elalish) [inline src/polygon/src/polygon.cpp:126]: Seems like there's not much point dumping this format now - can we update it to the format you now use in the `txt` files?

File: test/polygon_fuzz.cpp [hunk 3]
```diff
@@ -103,21 +104,39 @@ void TestPoly(Polygons polys, int _unused, float precision = -1.0) {
...
 std::vector<TestCase> SeedProvider() {
+  std::string file = __FILE__;
+  std::string dir = file.substr(0, file.rfind('/'));
+  auto f = std::ifstream(dir + "/polygons/" + "polygon_corpus.txt");
+  // for each test:
+  //   test name, expectedNumTri, precision, num polygons
+  //   for each polygon:
+  //     num points
+  //     for each vertex:
+  //       x coord, y coord
```
Comment (elalish) [inline test/polygon_fuzz.cpp:116]: 👍

**Review by elalish — CHANGES_REQUESTED** [review]
Thanks, that's a significant improvement! Remember to undo my change to `manifold.yml` that's skipping the zebra tests on windows.

File: test/polygons/conversion.py (hunk 1 above)
Comment (pca006132) [reply test/polygons/conversion.py:2]: We don't need this if we dump the new format directly.

File: src/polygon/src/polygon.cpp (hunk 2 above)
Comment (pca006132) [reply src/polygon/src/polygon.cpp:126]: Sure. My major concern is that the new format can be a bit hard to read (as I want to keep the parser simple), and the old c++ format is more self-explaining.

**Comment by pca006132** [comment]
I have no idea why the formatting thing is now failing...

File: src/polygon/src/polygon.cpp [hunk 4]
```diff
@@ -144,12 +144,12 @@ void PrintFailure(const std::exception &e, const PolygonsIdx &polys,
   std::cout << "-----------------------------------" << std::endl;
   std::cout << "Triangulation failed! Precision = " << precision << std::endl;
   std::cout << e.what() << std::endl;
-  if (triangles.size() > 1000) {
+  if (triangles.size() > 1000 && !PolygonParams().verbose) {
```
Comment (elalish) [inline src/polygon/src/polygon.cpp:147]: 👍

**Review by elalish — APPROVED** [review]
Thanks!

**Pull request merged**